time time
uint16 code
bool pressed
